From 3eba80300859ed86f6f23cc2b60ef9426b57a0cf Mon Sep 17 00:00:00 2001
Message-ID: <3eba80300859ed86f6f23cc2b60ef9426b57a0cf.1787144196.git.jdenemar@redhat.com>
From: Peter Krempa <pkrempa@redhat.com>
Date: Mon, 20 Jul 2026 13:43:26 +0200
Subject: [PATCH] qemu: capabilities: Retire QEMU_CAPS_SET_ACTION
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The capability is no longer used because all qemu versions we support do
have the 'set-action' command.

Signed-off-by: Peter Krempa <pkrempa@redhat.com>
Reviewed-by: Ján Tomko <jtomko@redhat.com>
(cherry picked from commit 01cb769a183e40a3f820ed9f949cf2d8b8cc3498)

https://redhat.atlassian.net/browse/RHEL-242545
---
 src/qemu/qemu_capabilities.c                               | 3 +--
 src/qemu/qemu_capabilities.h                               | 2 +-
 tests/qemucapabilitiesdata/caps_10.0.0_aarch64.xml         | 1 -
 tests/qemucapabilitiesdata/caps_10.0.0_ppc64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_10.0.0_s390x.xml           | 1 -
 tests/qemucapabilitiesdata/caps_10.0.0_x86_64+amdsev.xml   | 1 -
 tests/qemucapabilitiesdata/caps_10.0.0_x86_64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_10.1.0_s390x.xml           | 1 -
 tests/qemucapabilitiesdata/caps_10.1.0_x86_64+inteltdx.xml | 1 -
 tests/qemucapabilitiesdata/caps_10.1.0_x86_64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_10.2.0_aarch64.xml         | 1 -
 tests/qemucapabilitiesdata/caps_10.2.0_x86_64+mshv.xml     | 1 -
 tests/qemucapabilitiesdata/caps_10.2.0_x86_64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_11.0.0_aarch64.xml         | 1 -
 tests/qemucapabilitiesdata/caps_11.0.0_s390x.xml           | 1 -
 tests/qemucapabilitiesdata/caps_11.0.0_x86_64+sgx.xml      | 1 -
 tests/qemucapabilitiesdata/caps_11.0.0_x86_64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_11.1.0_aarch64.xml         | 1 -
 tests/qemucapabilitiesdata/caps_11.1.0_x86_64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_7.2.0_ppc.xml              | 1 -
 tests/qemucapabilitiesdata/caps_7.2.0_x86_64+hvf.xml       | 1 -
 tests/qemucapabilitiesdata/caps_7.2.0_x86_64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_8.0.0_x86_64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_8.1.0_s390x.xml            | 1 -
 tests/qemucapabilitiesdata/caps_8.1.0_x86_64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_8.2.0_aarch64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_8.2.0_armv7l.xml           | 1 -
 tests/qemucapabilitiesdata/caps_8.2.0_loongarch64.xml      | 1 -
 tests/qemucapabilitiesdata/caps_8.2.0_s390x.xml            | 1 -
 tests/qemucapabilitiesdata/caps_8.2.0_x86_64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_9.0.0_sparc.xml            | 1 -
 tests/qemucapabilitiesdata/caps_9.0.0_x86_64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_9.1.0_riscv64.xml          | 1 -
 tests/qemucapabilitiesdata/caps_9.1.0_s390x.xml            | 1 -
 tests/qemucapabilitiesdata/caps_9.1.0_x86_64.xml           | 1 -
 tests/qemucapabilitiesdata/caps_9.2.0_aarch64+hvf.xml      | 1 -
 tests/qemucapabilitiesdata/caps_9.2.0_s390x.xml            | 1 -
 tests/qemucapabilitiesdata/caps_9.2.0_x86_64+amdsev.xml    | 1 -
 tests/qemucapabilitiesdata/caps_9.2.0_x86_64.xml           | 1 -
 39 files changed, 2 insertions(+), 40 deletions(-)

diff --git a/src/qemu/qemu_capabilities.c b/src/qemu/qemu_capabilities.c
index c60ca02d97..f46c3b2759 100644
--- a/src/qemu/qemu_capabilities.c
+++ b/src/qemu/qemu_capabilities.c
@@ -643,7 +643,7 @@ VIR_ENUM_IMPL(virQEMUCaps,
               "confidential-guest-support", /* QEMU_CAPS_MACHINE_CONFIDENTAL_GUEST_SUPPORT */
               "query-display-options", /* X_QEMU_CAPS_QUERY_DISPLAY_OPTIONS */
               "s390-pv-guest", /* QEMU_CAPS_S390_PV_GUEST */
-              "set-action", /* QEMU_CAPS_SET_ACTION */
+              "set-action", /* X_QEMU_CAPS_SET_ACTION */
               "virtio-blk.queue-size", /* QEMU_CAPS_VIRTIO_BLK_QUEUE_SIZE */
 
               /* 410 */
@@ -1291,7 +1291,6 @@ struct virQEMUCapsStringFlags virQEMUCapsCommands[] = {
     { "query-cpu-definitions", QEMU_CAPS_QUERY_CPU_DEFINITIONS },
     { "query-cpu-model-baseline", QEMU_CAPS_QUERY_CPU_MODEL_BASELINE },
     { "query-cpu-model-comparison", QEMU_CAPS_QUERY_CPU_MODEL_COMPARISON },
-    { "set-action", QEMU_CAPS_SET_ACTION },
     { "query-dirty-rate", QEMU_CAPS_QUERY_DIRTY_RATE },
     { "sev-inject-launch-secret", QEMU_CAPS_SEV_INJECT_LAUNCH_SECRET },
     { "calc-dirty-rate", QEMU_CAPS_CALC_DIRTY_RATE },
diff --git a/src/qemu/qemu_capabilities.h b/src/qemu/qemu_capabilities.h
index 28085144a2..3fcee1dbe2 100644
--- a/src/qemu/qemu_capabilities.h
+++ b/src/qemu/qemu_capabilities.h
@@ -617,7 +617,7 @@ typedef enum { /* virQEMUCapsFlags grouping marker for syntax-check */
     QEMU_CAPS_MACHINE_CONFIDENTAL_GUEST_SUPPORT, /* -machine confidential-guest-support */
     X_QEMU_CAPS_QUERY_DISPLAY_OPTIONS, /* 'query-display-options' qmp command present */
     QEMU_CAPS_S390_PV_GUEST, /* -object s390-pv-guest,... */
-    QEMU_CAPS_SET_ACTION, /* 'set-action' QMP command */
+    X_QEMU_CAPS_SET_ACTION, /* 'set-action' QMP command */
     QEMU_CAPS_VIRTIO_BLK_QUEUE_SIZE, /* virtio-blk-*.queue-size */
 
     /* 410 */
diff --git a/tests/qemucapabilitiesdata/caps_10.0.0_aarch64.xml b/tests/qemucapabilitiesdata/caps_10.0.0_aarch64.xml
index b2a0872853..48629e494d 100644
--- a/tests/qemucapabilitiesdata/caps_10.0.0_aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_10.0.0_aarch64.xml
@@ -109,7 +109,6 @@
   <flag name='acpi-index'/>
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.0.0_ppc64.xml b/tests/qemucapabilitiesdata/caps_10.0.0_ppc64.xml
index 0b9613b921..70bf98c404 100644
--- a/tests/qemucapabilitiesdata/caps_10.0.0_ppc64.xml
+++ b/tests/qemucapabilitiesdata/caps_10.0.0_ppc64.xml
@@ -117,7 +117,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.0.0_s390x.xml b/tests/qemucapabilitiesdata/caps_10.0.0_s390x.xml
index 96eadffcec..25d73b6e1d 100644
--- a/tests/qemucapabilitiesdata/caps_10.0.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_10.0.0_s390x.xml
@@ -81,7 +81,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.0.0_x86_64+amdsev.xml b/tests/qemucapabilitiesdata/caps_10.0.0_x86_64+amdsev.xml
index 945f7364db..53fd51122c 100644
--- a/tests/qemucapabilitiesdata/caps_10.0.0_x86_64+amdsev.xml
+++ b/tests/qemucapabilitiesdata/caps_10.0.0_x86_64+amdsev.xml
@@ -145,7 +145,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.0.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_10.0.0_x86_64.xml
index 5cd3828f52..ad3fe67638 100644
--- a/tests/qemucapabilitiesdata/caps_10.0.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_10.0.0_x86_64.xml
@@ -145,7 +145,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.1.0_s390x.xml b/tests/qemucapabilitiesdata/caps_10.1.0_s390x.xml
index 584a1b304c..0d46b3b20e 100644
--- a/tests/qemucapabilitiesdata/caps_10.1.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_10.1.0_s390x.xml
@@ -82,7 +82,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.1.0_x86_64+inteltdx.xml b/tests/qemucapabilitiesdata/caps_10.1.0_x86_64+inteltdx.xml
index 6dd00c360f..242215f7a8 100644
--- a/tests/qemucapabilitiesdata/caps_10.1.0_x86_64+inteltdx.xml
+++ b/tests/qemucapabilitiesdata/caps_10.1.0_x86_64+inteltdx.xml
@@ -128,7 +128,6 @@
   <flag name='acpi-index'/>
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.1.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_10.1.0_x86_64.xml
index e7a0818697..a2faddcced 100644
--- a/tests/qemucapabilitiesdata/caps_10.1.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_10.1.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.2.0_aarch64.xml b/tests/qemucapabilitiesdata/caps_10.2.0_aarch64.xml
index f3d06927f6..3bbba5ddb1 100644
--- a/tests/qemucapabilitiesdata/caps_10.2.0_aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_10.2.0_aarch64.xml
@@ -122,7 +122,6 @@
   <flag name='input-linux'/>
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.2.0_x86_64+mshv.xml b/tests/qemucapabilitiesdata/caps_10.2.0_x86_64+mshv.xml
index 54018639ad..530c0366ea 100644
--- a/tests/qemucapabilitiesdata/caps_10.2.0_x86_64+mshv.xml
+++ b/tests/qemucapabilitiesdata/caps_10.2.0_x86_64+mshv.xml
@@ -135,7 +135,6 @@
   <flag name='acpi-index'/>
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_10.2.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_10.2.0_x86_64.xml
index d5cc279e13..b2cd261b99 100644
--- a/tests/qemucapabilitiesdata/caps_10.2.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_10.2.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_11.0.0_aarch64.xml b/tests/qemucapabilitiesdata/caps_11.0.0_aarch64.xml
index ca6def23dc..d77884f401 100644
--- a/tests/qemucapabilitiesdata/caps_11.0.0_aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_11.0.0_aarch64.xml
@@ -122,7 +122,6 @@
   <flag name='input-linux'/>
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_11.0.0_s390x.xml b/tests/qemucapabilitiesdata/caps_11.0.0_s390x.xml
index 99fccc9739..07ee859318 100644
--- a/tests/qemucapabilitiesdata/caps_11.0.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_11.0.0_s390x.xml
@@ -82,7 +82,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_11.0.0_x86_64+sgx.xml b/tests/qemucapabilitiesdata/caps_11.0.0_x86_64+sgx.xml
index bac7951f73..108f4b0b86 100644
--- a/tests/qemucapabilitiesdata/caps_11.0.0_x86_64+sgx.xml
+++ b/tests/qemucapabilitiesdata/caps_11.0.0_x86_64+sgx.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_11.0.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_11.0.0_x86_64.xml
index 8724ce5b56..29fdaf20d8 100644
--- a/tests/qemucapabilitiesdata/caps_11.0.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_11.0.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_11.1.0_aarch64.xml b/tests/qemucapabilitiesdata/caps_11.1.0_aarch64.xml
index 306f2981c1..56fbb80028 100644
--- a/tests/qemucapabilitiesdata/caps_11.1.0_aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_11.1.0_aarch64.xml
@@ -122,7 +122,6 @@
   <flag name='input-linux'/>
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_11.1.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_11.1.0_x86_64.xml
index 0c020702b4..1d1eef21ff 100644
--- a/tests/qemucapabilitiesdata/caps_11.1.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_11.1.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_7.2.0_ppc.xml b/tests/qemucapabilitiesdata/caps_7.2.0_ppc.xml
index bc5953f43e..bf67b3d295 100644
--- a/tests/qemucapabilitiesdata/caps_7.2.0_ppc.xml
+++ b/tests/qemucapabilitiesdata/caps_7.2.0_ppc.xml
@@ -109,7 +109,6 @@
   <flag name='input-linux'/>
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='piix4.acpi-root-pci-hotplug'/>
diff --git a/tests/qemucapabilitiesdata/caps_7.2.0_x86_64+hvf.xml b/tests/qemucapabilitiesdata/caps_7.2.0_x86_64+hvf.xml
index 425c608863..f40d77baa5 100644
--- a/tests/qemucapabilitiesdata/caps_7.2.0_x86_64+hvf.xml
+++ b/tests/qemucapabilitiesdata/caps_7.2.0_x86_64+hvf.xml
@@ -146,7 +146,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_7.2.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_7.2.0_x86_64.xml
index 93fe36f48c..8d44ed71e0 100644
--- a/tests/qemucapabilitiesdata/caps_7.2.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_7.2.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.0.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_8.0.0_x86_64.xml
index 592df7049b..fcbb68e415 100644
--- a/tests/qemucapabilitiesdata/caps_8.0.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_8.0.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.1.0_s390x.xml b/tests/qemucapabilitiesdata/caps_8.1.0_s390x.xml
index 2c540f7f30..65f73d1af8 100644
--- a/tests/qemucapabilitiesdata/caps_8.1.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_8.1.0_s390x.xml
@@ -77,7 +77,6 @@
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.1.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_8.1.0_x86_64.xml
index b29325a465..c730dd7174 100644
--- a/tests/qemucapabilitiesdata/caps_8.1.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_8.1.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.2.0_aarch64.xml b/tests/qemucapabilitiesdata/caps_8.2.0_aarch64.xml
index e026abfd96..1f0f68092d 100644
--- a/tests/qemucapabilitiesdata/caps_8.2.0_aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_8.2.0_aarch64.xml
@@ -115,7 +115,6 @@
   <flag name='acpi-index'/>
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.2.0_armv7l.xml b/tests/qemucapabilitiesdata/caps_8.2.0_armv7l.xml
index d3af74bcb3..254ec27545 100644
--- a/tests/qemucapabilitiesdata/caps_8.2.0_armv7l.xml
+++ b/tests/qemucapabilitiesdata/caps_8.2.0_armv7l.xml
@@ -120,7 +120,6 @@
   <flag name='input-linux'/>
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.2.0_loongarch64.xml b/tests/qemucapabilitiesdata/caps_8.2.0_loongarch64.xml
index 640438623a..95eb2dfa21 100644
--- a/tests/qemucapabilitiesdata/caps_8.2.0_loongarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_8.2.0_loongarch64.xml
@@ -108,7 +108,6 @@
   <flag name='acpi-index'/>
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.2.0_s390x.xml b/tests/qemucapabilitiesdata/caps_8.2.0_s390x.xml
index a254e16741..45ab66ec29 100644
--- a/tests/qemucapabilitiesdata/caps_8.2.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_8.2.0_s390x.xml
@@ -77,7 +77,6 @@
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_8.2.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_8.2.0_x86_64.xml
index 57aa43adab..b46505d24b 100644
--- a/tests/qemucapabilitiesdata/caps_8.2.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_8.2.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.0.0_sparc.xml b/tests/qemucapabilitiesdata/caps_9.0.0_sparc.xml
index c2710ecab1..44ddfb0f76 100644
--- a/tests/qemucapabilitiesdata/caps_9.0.0_sparc.xml
+++ b/tests/qemucapabilitiesdata/caps_9.0.0_sparc.xml
@@ -37,7 +37,6 @@
   <flag name='rotation-rate'/>
   <flag name='input-linux'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
   <flag name='query-dirty-rate'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.0.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_9.0.0_x86_64.xml
index 3b4aaa202c..2f361bd1c8 100644
--- a/tests/qemucapabilitiesdata/caps_9.0.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_9.0.0_x86_64.xml
@@ -147,7 +147,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.1.0_riscv64.xml b/tests/qemucapabilitiesdata/caps_9.1.0_riscv64.xml
index b5ca67b445..f04b2fd302 100644
--- a/tests/qemucapabilitiesdata/caps_9.1.0_riscv64.xml
+++ b/tests/qemucapabilitiesdata/caps_9.1.0_riscv64.xml
@@ -113,7 +113,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.1.0_s390x.xml b/tests/qemucapabilitiesdata/caps_9.1.0_s390x.xml
index 02c8e6f641..990fe255f6 100644
--- a/tests/qemucapabilitiesdata/caps_9.1.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_9.1.0_s390x.xml
@@ -81,7 +81,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.1.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_9.1.0_x86_64.xml
index 65d279f5ea..07d7562c84 100644
--- a/tests/qemucapabilitiesdata/caps_9.1.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_9.1.0_x86_64.xml
@@ -145,7 +145,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.2.0_aarch64+hvf.xml b/tests/qemucapabilitiesdata/caps_9.2.0_aarch64+hvf.xml
index 8c777c121e..c9e85ff9b3 100644
--- a/tests/qemucapabilitiesdata/caps_9.2.0_aarch64+hvf.xml
+++ b/tests/qemucapabilitiesdata/caps_9.2.0_aarch64+hvf.xml
@@ -87,7 +87,6 @@
   <flag name='rotation-rate'/>
   <flag name='acpi-index'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='netdev.json'/>
   <flag name='query-dirty-rate'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.2.0_s390x.xml b/tests/qemucapabilitiesdata/caps_9.2.0_s390x.xml
index 135812b527..a8f830076e 100644
--- a/tests/qemucapabilitiesdata/caps_9.2.0_s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_9.2.0_s390x.xml
@@ -81,7 +81,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='confidential-guest-support'/>
   <flag name='s390-pv-guest'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='memory-backend-file.reserve'/>
   <flag name='netdev.json'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.2.0_x86_64+amdsev.xml b/tests/qemucapabilitiesdata/caps_9.2.0_x86_64+amdsev.xml
index af4b8bbc3e..f67222578d 100644
--- a/tests/qemucapabilitiesdata/caps_9.2.0_x86_64+amdsev.xml
+++ b/tests/qemucapabilitiesdata/caps_9.2.0_x86_64+amdsev.xml
@@ -146,7 +146,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
diff --git a/tests/qemucapabilitiesdata/caps_9.2.0_x86_64.xml b/tests/qemucapabilitiesdata/caps_9.2.0_x86_64.xml
index 933698d99f..efd3eeb9c7 100644
--- a/tests/qemucapabilitiesdata/caps_9.2.0_x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_9.2.0_x86_64.xml
@@ -145,7 +145,6 @@
   <flag name='virtio-gpu-gl-pci'/>
   <flag name='virtio-vga-gl'/>
   <flag name='confidential-guest-support'/>
-  <flag name='set-action'/>
   <flag name='virtio-blk.queue-size'/>
   <flag name='virtio-mem-pci'/>
   <flag name='memory-backend-file.reserve'/>
-- 
2.55.0
